src/Rings/groebner.jl

```diff
@@ -15,34 +15,40 @@ accessing `I.gb.S`.
 
 # Examples
 ```jldoctest
-julia> R,(x,y) = PolynomialRing(QQ, ["x","y"], ordering=:degrevlex)
+julia> R,(x,y) = PolynomialRing(QQ, ["x","y"])
 (Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])
 
 julia> I = ideal([x*y-3*x,y^3-2*x^2*y])
 ideal(x*y - 3*x, -2*x^2*y + y^3)
 
-julia> Oscar.groebner_assure(I)
-3-element Vector{fmpq_mpoly}:
- x*y - 3*x
- y^3 - 6*x^2
- x^3 - 9//2*x
+julia> Oscar.groebner_assure(I, degrevlex(gens(R)));
 
-julia> I.gb
-Oscar.BiPolyArray{fmpq_mpoly}(fmpq_mpoly[x*y - 3*x, y^3 - 6*x^2, x^3 - 9//2*x], Singular ideal over Singular Polynomial Ring (QQ),(x,y),(dp(2),C) with generators (x*y - 3*x, y^3 - 6*x^2, x^3 - 9//2*x), Multivariate Polynomial Ring in x, y over Rational Field, Singular Polynomial Ring (QQ),(x,y),(dp(2),C), true, #undef, false)
+julia> I.gb[degrevlex(gens(R))]
+Oscar.BiPolyArray{fmpq_mpoly}(fmpq_mpoly[#undef, #undef, #undef], Singular ideal over Singular Polynomial Ring (QQ),(x,y),(dp(2),C) with generators (x*y - 3*x, y^3 - 6*x^2, 2*x^3 - 9*x), Multivariate Polynomial Ring in x, y over Rational Field, Singular Polynomial Ring (QQ),(x,y),(dp(2),C), true, #undef, true)
 ```
 """
-function groebner_assure(I::MPolyIdeal; complete_reduction::Bool = false)
-  if !isdefined(I, :gb)
-    singular_assure(I)
-    i = Singular.std(I.gens.S; complete_reduction = complete_reduction)
-    I.gb = BiPolyArray(base_ring(I), i)
-    I.gb.isGB = true
-    I.gb.O = [I.gb.Ox(x) for x = gens(I.gb.S)]
-  end
+function groebner_assure(I::MPolyIdeal, complete_reduction::Bool = false)
+    if isempty(I.gb)
+        drl = degrevlex(gens(base_ring(I)))
+        I.gb[drl] = groebner_assure(I, drl, complete_reduction)
+        G = I.gb[drl]
+    else
+        G = collect(values(I.gb))[1]
+    end
+    return G
+end
+
+function groebner_assure(I::MPolyIdeal, ordering::MonomialOrdering,  complete_reduction::Bool = false)
+    if get(I.gb, ordering, -1) == -1
```

Comment:
```suggestion
    if !haskey(I.gb, ordering)
```
Could also be rewritten as
```
get!(I.gb, ordering) do _compute_groebner_basis(I.gens, ordering, complete_reduction) end
```
if you want to avoid the double lookup (should not matter here, just saying).